mysite/psa/templatetags/social_tags.py

```diff
@@ -0,0 +1,141 @@
+import re
+
+from django import template
+from django.conf import settings
+from collections import defaultdict
+
+from social.backends.utils import load_backends
+from social.backends.oauth import OAuthAuth
+
+from psa.models import SecondaryEmail
+
+
+register = template.Library()
+name_re = re.compile(r'([^O])Auth')
+
+
+@register.filter
+def backend_name(backend):
+    name = backend.__class__.__name__
+    name = name.replace('OAuth2', '')
+    name = name.replace('OAuth1', '')
+    name = name.replace('OAuth', '')
+    name = name.replace('Auth', '')
+    name = name.replace('OpenId', '')
+    name = name.replace('Sandbox', '')
+    name = name_re.sub(r'\1 Auth', name)
+    return name
+
+
+@register.filter
+def backend_class(backend):
+    return backend.name.replace('-', ' ')
+
+
+@register.filter
+def presentation_name(name):
+    return {
+        'stackoverflow': 'Stack-overflow',
+        'google-oauth': 'Google',
+        'google-oauth2': 'Google',
+        'google-openidconnect': 'Google',
+        'yahoo-oauth': 'Yahoo',
+        'facebook-app': 'Facebook',
+        'email': 'Email',
+        'vimeo': 'Vimeo-square',
+        'linkedin-oauth2': 'Linkedin',
+        'vk-oauth2': 'Vk',
+        'live': 'Windows',
+        'username': 'User',
+    }.get(name, name)
+
+
+@register.filter
+def icon_name(name):
+    return {
+        'stackoverflow': 'stack-overflow',
+        'google-oauth': 'google',
+        'google-oauth2': 'google',
+        'google-openidconnect': 'google',
+        'yahoo-oauth': 'yahoo',
+        'facebook-app': 'facebook',
+        'email': 'envelope',
+        'vimeo': 'vimeo-square',
+        'linkedin-oauth2': 'linkedin',
+        'vk-oauth2': 'vk',
+        'live': 'windows',
+        'username': 'user',
+    }.get(name, name)
+
+
+@register.filter
+def social_backends(backends):
+    backends = [(name, backend) for name, backend in backends.items()
+                if name not in ['username', 'email']]
+    backends.sort(key=lambda b: b[0])
+
+    return [backends[n:n + 10] for n in range(0, len(backends), 10)]
```

Comment:
> What does that n+0 means?

This is pagination in case of more then 10 social providers. I will add this in comment